pkg/ddl/constraint.go

```diff
@@ -139,6 +139,21 @@ func checkAddCheckConstraint(t *meta.Meta, job *model.Job) (*model.DBInfo, *mode
 		}
 		// if not, that means constraint was in intermediate state.
 	}
+
+	tblInfos, err := t.ListTables(schemaID)
+	if err != nil {
+		return nil, nil, nil, nil, err
+	}
+
+	for _, tbl := range tblInfos {
+		if tbl.ID != tblInfo.ID {
+			if constraintInfo := tbl.FindConstraintInfoByName(constraintInfo1.Name.L); constraintInfo != nil {
+				job.State = model.JobStateCancelled
+				return nil, nil, nil, nil, infoschema.ErrCheckConstraintDupName.GenWithStackByArgs(constraintInfo1.Name.L)
+			}
+		}
+	}
+
```

Comment:
Can we use `checkConstraintNamesNotExists` instead?
```suggestion

	err = checkConstraintNamesNotExists(t, schemaID, []*model.ConstraintInfo{constraintInfo1})
	if err != nil {
		job.State = model.JobStateCancelled
		return nil, nil, nil, nil, err
	}

```

Reply:
fix: https://github.com/pingcap/tidb/pull/48062/commits/f4720bb7dc4c089a18d36235dcccd5577b5b8e12